### PR TITLE
Server: Add registries

### PR DESCRIPTION
**Pull request by mattayes**
# Changes

* Add actorMakeRegistry and mailboxRegistry to manage registry CRUD.
* Add tests (coverage ≈75%).
* Miscellaneous cleanup.

I'm not positive this will solve our issue. It should reduce the likelihood of locking-related issues, though.

### CODE REVIEW

**Review by vitaminmoo — APPROVED** [review]

File: actor.go [hunk 1]
```diff
@@ -37,7 +43,28 @@ func NewActorStart(name string, v ...interface{}) *ActorStart {
...
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	m, found = r.r[name]
+	return
+}
+
+func (r *makeActorRegistry) Set(name string, m MakeActor) (update bool) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	_, update = r.r[name]
```
Comment (epsniff) [inline actor.go:67]: How did this even compile? 
```suggestion
```

Comment (epsniff) [reply actor.go:67]: We may need to check travis, maybe we're using an old version of Go for this repo.

Comment (epsniff) [reply actor.go:67]: nm, I didn't notice that you're using a named var in the return.

File: mailbox.go [hunk 2]
```diff
@@ -18,28 +18,85 @@ var (
...
+
+// Size of the registry.
+func (r *mailboxRegistry) Size() int {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return len(r.r)
+}
+
+// R returns the underlying registry.
+// It does _not_ return a copy.
```
Comment (epsniff) [inline mailbox.go:69]: What do you mean?  It looks like it DOES return a copy of the mailbox map... 

I assume this is for testing only, so I'd add that comment.

File: server.go [hunk 3]
```diff
@@ -487,3 +467,12 @@ func (s *Server) logf(format string, v ...interface{}) {
 		s.cfg.Logger.Printf(format, v...)
 	}
 }
+
+func (s *Server) isServing() error {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	if !s.serving {
```
Comment (epsniff) [inline server.go:474]: I don't see where this is ever set to false?

**Review by epsniff — APPROVED** [review]
After you fix the serving flag comment, I approve.

File: mailbox.go (hunk 2 above)
Comment (mattayes) [reply mailbox.go:69]: @epsniff Modifying the map returned here modifying the map stored on the struct: https://go.dev/play/p/gBap9sul28C. That's what I meant by not a copy: if the returned map were a copy you could change it without impacting the struct's map.

File: server.go (hunk 3 above)
Comment (mattayes) [reply server.go:474]: @epsniff The zero value for booleans is `false`, so we don't need to explicitly set it.

**Review by ajroetker — APPROVED** [review]

File: mailbox.go (hunk 2 above)
Comment (epsniff) [reply mailbox.go]: I'd just go ahead and produce a copy of the map and make the func safe.   There isn't a performance reason to not copy it.  This function is called enough to try doing anything fancy.   It's better to no have a future foot gun.

File: server.go (hunk 3 above)
Comment (epsniff) [reply server.go:474]: Oh, I assumed this func was used during the Stop part of the life cycle and was used to indicate the system was shutting down.

Comment (mattayes) [reply server.go:474]: No, it’s used in NewMailbox() to ensure unstarted servers don’t create mailboxes.

**Pull request merged**